testRearrangeCarGroup8.py

```diff
@@ -0,0 +1,118 @@
+import unittest
+import numpy as np
+from RearrangeCarGroup8 import RearrangeCars
+
+class TestRearrangeCars(unittest.TestCase):
+
+    '''Unittests for RearrangeCars class and promised fuctionality
+    '''
+
+    def setUp(self):
+
+        '''Basic setting that will be used in several tests
+        '''
+        
+        self.initial_car_order = [1, 2, 0, 3]
+        self.final_car_order = [3, 1, 2, 0]
+        self.rc = RearrangeCars(list(self.initial_car_order), list(self.final_car_order))

```

Comment:
Yes, that was what I was afraid of, because I think it happened before... Previously, when the code in `RearrangeCarGroup8`contains the line `self.current_car_order = self.initial_car_order`, instead of `self.current_car_order = list(self.initial_car_order)`, these two variables refer to the same object and `self.intial_car_order` was modified. However, when I checked whether `self.rc.initial_car_order` was the same as `self.initial_car_order`, the test passed like a charm. When I printed them out, however, both had changed and became the same as `self.rc.current_car_order`, since they all pointed to the same object. That was why I listed them... OMG, hope this makes sense XD